bin/common.c

```diff
@@ -263,29 +264,40 @@ static int s2n_setup_external_psk(struct s2n_psk **psk, char *params)
     GUARD_EXIT_NULL(psk);
     GUARD_EXIT_NULL(params);
 
+    DEFER_CLEANUP(struct s2n_stuffer psk_params = { 0 }, s2n_stuffer_free);
+    DEFER_CLEANUP(struct s2n_stuffer token = { 0 }, s2n_stuffer_free);
+    POSIX_GUARD(s2n_stuffer_alloc_ro_from_string(&psk_params, params));
+    POSIX_GUARD(s2n_stuffer_growable_alloc(&token, 0));
     size_t token_idx = 0;
-    for (char *token = strtok(params, ","); token != NULL; token = strtok(NULL, ","), token_idx++) {
+
```

Comment:
Copying the params before you tokenize them should work, but you can't use the stuffer methods.

Alternatively, only tokenize the input once when the server starts up instead of on every connection.

Reply:
I went with copying the params